R/authenticate.R

```diff
@@ -15,21 +16,31 @@
 #' @export
 #' @examples
 #' \dontrun{
-#' # Use with caution, don't let prying eyes see your password.
-#' authenticate("username", "password")
-#' # Instead, you could store it in an environment variable.
-#' authenticate("username", Sys.getenv("API_PASSWORD"))
+#' authenticate() # prompts for username and password (and OTP if configured)
+#' authenticate("username") # prompts for password (and OTP if configured)
 #'
-#' # If the password is omitted and you're running in RStudio or the getPass
-#' # library is installed, a prompt will be displayed.
-#' authenticate("username")
+#' # For non-interactive sessions, use an environment variable or other means to
+#' # avoid saving your password in plaintext.
+#' authenticate("username", Sys.getenv("API_PASSWORD"))
 #' }
-authenticate <- function(username, password = NA, otp = NA) {
+authenticate <- function(username = NA, password = NA, otp = NA) {
+  if (is.na(username)) {
+    if (rstudioapi::isAvailable()) {
+      username <- rstudioapi::showPrompt("Username", "Please enter your username or email")
+    } else if (interactive()) {
+      username <- readline("Please enter your username or email: ")
+    } else {
+      stop("Username is required, but was not provided and no interactive prompts are available")
+    }
+  }
+
   if (is.na(password)) {
-    if (requireNamespace("getPass")) {
+    if (requireNamespace("getPass", quietly=TRUE)) {
```

Comment:
just a nit from the linter
```suggestion
    if (requireNamespace("getPass", quietly = TRUE)) {
```